hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -118,6 +120,9 @@ static String getScope(Configuration conf, String clusterId, Object connectionOb
 
   private static final String CNT_BASE = "rpcCount_";
   private static final String FAILURE_CNT_BASE = "rpcFailureCount_";
+  private static final String TIMEOUT_CNT_BASE = "rpcExceptionCallTimeout_";
+  private static final String REMOTE_CNT_BASE = "rpcExceptionRemote_";
```

Comment:
A remote exception can be caused by many things. Should this be more granular? At least consider differentiating between non retryable DoNotRetryIOExceptions and the other retryable IOExceptions. So that would be two counters here for remote.

Reply:
Or differentiate between all the **timeout** caused exceptions and all of the **error** caused exceptions.